juju/client/connector.py

```diff
@@ -73,6 +78,10 @@ async def connect(self, **kwargs):
                 await self._connection.close()
             self._connection = await Connection.connect(**kwargs)
 
+        # Check if we support the target controller
+        if not self._connection.info['server-version'].startswith(SUPPORTED_JUJU_API_PREFIX):
```

Comment:
This will also break for any version that's >=3.1, so I'd do `SUPPORTED_JUJU_API_PREFIX = "3."`

Reply:
We will update it once we are sure we can support 3.1